### ISSUE TITLE
Nuget Respect `repositoryUrls` in config.js

### ISSUE THREAD

**Issue by BenB196** [renovatebot/renovate#21091]
### What would you like Renovate to be able to do?

I would like it so that if I define `repositoryUrls` for `nuget` in config.js, like:

```js
module.exports = {
  "nuget": {
    "registryUrls": [
      "https://gitlab.internal.com/api/v4/groups/1/-/packages/nuget/index.json",
      "https://api.nuget.org/v3/index.json"
    ]
  },
  "hostRules": [
    "matchHost": "https://gitlab.internal.com/api/v4/groups/1/-/packages/nuget/",
    "hostType": "nuget",
    "username": "user",
    "password": "password"
  ]
}
```

That Renovate respects this definition. This is somewhat of a necro of https://github.com/renovatebot/renovate/issues/13591, but I'm going to try and provide a (somewhat) valid use-case for why I think this might be a useful feature.

---

### Use-case

Today, I have a relatively large number of dotnet projects/repositories. These projects don't leverage `nuget.conf` because of some functional issues. Today, either developers have local versions of the packages their referencing on their computer, or as part of CI pipelines, a `before_script` `dotnet nuget add source` is run before the actual execution. This means we can't really leverage Renovate with private repositories.

Being able to define nuget registryUrls would allow for our current use-pattern (while technically not ideal), but still being able to use custom nuget registries with Renovate

### If you have any ideas on how this should be implemented, please tell us here.

Looking at the previous issue on this, I think something like this might work:

https://github.com/renovatebot/renovate/blob/a6af88eb231a6f990b1f055c772228856425b5c5/lib/manager/nuget/artifacts.ts#L36-L38

```ts
const registries =
    (await getConfiguredRegistries(packageFileName, localDir)) ||
// inject registryUrls here, then || if registryUrls isn't set
    getDefaultRegistries();
```

The above would prioritize nuget.conf to keep that as the "desired" solution, while allowing for people who don't/can't use nuget.conf to be able to use use `registryUrls` instead, and then falling back to the hardcoded `getDefaultRegistries();`, if not set as well.

### Is this a feature you are interested in implementing yourself?

Maybe

**Comment by rarkins** [comment]
Please also look into whether `defaultRegistryUrls` would already work for your use case

**Comment by rarkins** [comment]
OK, let's look into why that is as a first preference

**Comment by viceice** [comment]
it looks like some information is missing on the issue description. you can already change the default nugget registry urls. i know that, because I'm doing that on some internal projects and don't get any warning.

i think you like renovate to use the configured registry urls to be used for artifacts generation ( to update the lock files)?

Please clarify the issue description.

**Comment by BenB196** [comment]
Well, this is a bit embarrassing. I took another go at using `defaultRegistryUrls`, and it does seem to in fact work. Looking on my end on why I thought it didn't initially, it looks like I was hitting `prHourlyLimit`, and when I saw the debug logs saying `skipped`, I had initially assumed that it was because `defaultRegistryUrls` wasn't working rather than `prHourlyLimit` causing a new PR to be skipped. Bumping `prHourlyLimit` too a much higher number I was able to get Renovate to open a PR for the custom package.

Thanks for the assistance here (and apologies for the noise)
